### DISCUSSION TITLE
Problem for plotting misorientation axes

### DISCUSSION THREAD

**Discussion by rtaufner** [mtex-toolbox/mtex#595] [Ask Anything] [answered]
Hi everyone, hope you all are safe and sound.

I've been using a script to generate ebsd-based information for crystals of plagioclase, clinopyroxene and amphibole. The scripts and codes used for all three are pretty much the same, and I change the crystal reference and symmetry accordingly. Everything works like a charm; however, when I run the code to calculate the misorientation axes for plagioclase, I keep getting the following error:
```
Error using S2FunHarmonicSym
Too many input arguments.

Error in S2FunHarmonic/symmetrise (line 63)
  sFs = S2FunHarmonicSym(sF.fhat, sym, varargin{:});

Error in S2FunHarmonic/subsref (line 17)
    [varargout{1:nargout}] = builtin('subsref',sF,s);

Error in Miller/calcDensity (line 43)
sF = sF.symmetrise(h.CS);

Error in vector3d/smooth (line 36)
    cdata = calcDensity(v(:),S2G,'halfwidth',5*degree,varargin{:});


Error in Miller/smooth (line 43)
[varargout{1:nargout}] = smooth@vector3d(m,varargin{:},sR,m.CS);

Error in vector3d/contourf (line 31)
  [varargout{1:nargout}] = v.smooth(data,'contours',10,'LineStyle','-','LineColor','k',varargin{:});


Error in vector3d/plot (line 101)
    [varargout{1:nargout}] = v.contourf(varargin{:});
```
This error message does not pop up on the screen when I use the scripts for clinopyroxene and amphibole, which work fine. What I realised is that the message appears when I use the command 'contourf' to generate the inverse pole figure. If I get rid of it, the IPF with dots appears on screen. However, I want to create the IPF with contours.
Following is code that I've been using to calculate the misorientation axes:
```matlab
%% Subset of grains >= 0 (Plagioclase);

ebsdp=ebsd('Plagioclase');
grains_selected_a = selected_grains(selected_grains.grainSize > 0);
plot (grains_selected_a.boundary, 'figSize', 'large')
hold on
odf_grains_selected_a = calcODF(ebsdp(grains_selected_a).orientations,'halfwidth',10*degree)

%% Pair Misorientation Angles-Axes - subset of grains > 0 - CRYSTAL COORDINATES - 2-10 - Contours

gB_plag_plag_a = selected_grains.boundary(grains_selected_a,'Plagioclase',grains_selected_a,'Plagioclase');
MDF_plag_plag_Uniform = uniformODF(ebsd('Plagioclase').CS,ebsd('Plagioclase').CS);
MDF_plag_plag_Uncorrelated = calcMDF(odf_grains_selected_a,odf_grains_selected_a);

gb_axis_plag_a = axis(gB_plag_plag_a.misorientation);
gb_angle_plag_a = angle(gB_plag_plag_a.misorientation)./ degree;
condition = gb_angle_plag_a >= 2 & gb_angle_plag_a < 10;
selected_angles = gb_angle_plag_a(condition);
selected_axes = gb_axis_plag_a(condition);
N_axes = length(selected_axes)

figure
plot(selected_axes,'contourf','antipodal','fundamentalRegion','resolution', 3*degree, 'minmax')
colorbar
hold on
annotate([Miller(1,0,0,ebsd('Plagioclase').CS,'uvw'),...
          Miller(0,1,0,ebsd('Plagioclase').CS,'uvw'),...
          Miller(0,0,1,ebsd('Plagioclase').CS,'uvw'),...
          Miller(1,1,0,ebsd('Plagioclase').CS,'uvw'),...
          Miller(-1,1,0,ebsd('Plagioclase').CS,'uvw'),...
          Miller(1,0,1,ebsd('Plagioclase').CS,'uvw')],...
  'all','labeled','BackgroundColor','w','FontSize',18)
```

Could anybody help me with that?
Thanks a lot.
Rhander.

**Comment by ralfHielscher** [comment]
Hi Rhander,

would it be possible for you to reduce the script a little bit and investigate where any why the error pops up.
Eventually you end up with a list of axes `selected_axes`

Is everything ok with them? what is their symmetry?

probably `selected_axes.calcDensity` gives you then an error.

Ralf.

**Reply by rtaufner** [reply]
Hi Ralf, 

Thank you for helping me.

I went through each line of the script again. The `selected_axes` seem to be fine. Their symmetry is "-1" (see below).
<img width="352" alt="Screen Shot 2020-11-26 at 09 15 45" src="https://user-images.githubusercontent.com/75056142/100350542-ce894780-2fc8-11eb-9689-1bb7eb678cd7.png">

I tried to calculate `selected_axes.calcDensity` as you suggested but it returns to the same error. 
The lines run perfectly until I reach the point to plot the `selected_axes` as contoured plots. For example, plotting the axes as dots gives me this IFP for the `selected axes`:
![IPF](https://user-images.githubusercontent.com/75056142/100350806-363f9280-2fc9-11eb-8318-cdce5b21a858.jpg)

What is odd is that I used this same script a few months ago to generate some contoured IPF's for plagioclase and it was working okay. Now I'm just trying to review all the data and this error insists to follow me. I'm using Matlab 2015b and Mtex 5.1.1. 

Thanks again, 
Rhander.

**Comment by ralfHielscher** [comment, marked as answer]
The code in `S2FunHarmonic/symmetrise` (line 63)
has a bug. Instead of
```
  sFs = S2FunHarmonicSym(sF.fhat, sym, varargin{:});
```
it should be 
```
  sFs = S2FunHarmonicSym(sF.fhat, sym);
```
This issue has been fixed in MTEX 5.2. two years ago (commit ab1a365ca59c00af1d5b281d208f1c817bf266b9)

Ralf.

**Reply by rtaufner** [reply]
Hi Ralf, 

I've replaced the line 63 with the new syntax and it's now looking good.

Thanks a lot for all your help.

Have a great day.
Rhander.
